pipeline_games/load_games.py

```diff
@@ -89,72 +89,97 @@ def get_existing_platform_data(mac_c, windows_c, linux_c, conn: connection, cach
         return cache[value]
 
 
-def add_to_genre_link_table(conn: connection, data: list) -> None:
+def get_all_game_genre_ids(conn: connection, data: list) -> list[tuple]:
+    """Returns all game_genre_ids for linking table"""
+    tuples = [tuple(x) for x in data.to_numpy()]
+    query = """SELECT game.game_id, genre.genre_id FROM genre CROSS JOIN game
+            WHERE game.app_id = %s AND genre.genre = %s AND user_generated = %s;"""
+    all_ids = []
+    for line in tuples:
+        with conn.cursor() as cur:
+            cur.execute(query, line)
+            game_genre = cur.fetchall()
+            all_ids.append(
+                (game_genre[0]['game_id'], game_genre[0]['genre_id'],
+                 game_genre[0]['game_id'], game_genre[0]['genre_id']))
+    return all_ids
+
+
+def get_all_publisher_game_ids(conn: connection, data: list) -> list[tuple]:
+    """Returns all game_publisher ids for linking table"""
+    tuples = [tuple(x) for x in data.to_numpy()]
+    query = """SELECT game.game_id, publisher.publisher_id FROM publisher CROSS JOIN game
+            WHERE game.app_id = %s AND publisher.publisher_name = %s;"""
+    all_ids = []
+    for line in tuples:
+        with conn.cursor() as cur:
+            cur.execute(query, line)
+            game_genre = cur.fetchall()
+            all_ids.append(
+                (game_genre[0]['game_id'], game_genre[0]['publisher_id'],
+                 game_genre[0]['game_id'], game_genre[0]['publisher_id']))
```

Comment:
not very familiar with your code but what's the need for 2 the same ones?

Reply:
I need a tuple with 4 values, two are repeated as that's how it is used in batch. If you have a look at the batch function it is then used in, each part of the tuple replaces where there is a %s symbol

Reply:
I see. Thanks :)